### PR TITLE
Document nerves_validated and nerves_autovalidate

### PR DESCRIPTION
**Pull request by fhunleth**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -172,35 +173,50 @@ run:
...
-Here's a simple alternative:
-
-1. After upgrading firmware, save that the next boot is the first one.
-1. On the reboot, if this is the first boot, record that the boot happened and
+Running this command manually is useful in development. Production use requires
+more work to protect against faulty upgrades.
+
+### Assisted firmware validation and automatic revert
+
+Some Nerves systems all your application to determine whether newly installed
```
Comment (mobileoverlord) [inline README.md]: > Some Nerves systems all your application

This doesn't sound right

File: README.md [hunk 2]
```diff
@@ -172,35 +173,50 @@ run:
...
+firmware reconnect to the firmware update server in 5 minutes?" If the answer to
+that question is negative, the device should reboot back to the previously
+running firmware.
+
+Nerves systems that support the feature work as follows:
+
+1. New firmware is installed in the normal manner. The `Nerves.Runtime.KV`
+   variable, `nerves_validated` is set to 0.
+2. The system reboots like normal.
+3. The application starts a five minute reboot timer (this need not be the
```
Comment (mobileoverlord) [inline README.md]: > this need not be the

This should be reworded

File: README.md [hunk 3]
```diff
@@ -172,35 +173,50 @@ run:
...
+   application, but it is easier to understand)
+4. The application attempts to make a connection to the firmware update server.
+5. On a good connection, the application sets `nerves_validated` to 1 and
+   cancels the reboot timer.
+6. On error, the reboot timer failing, or a hardware watchdog timeout, the
+   system reboots. The bootloader reverts to the previous firmware.
+
+To use this feature, the `nerves_autovalidate` variable must be set to 0. This
+can be done at device provisioning time when other keys (like the serial number)
+are set or a custom `fwup.conf` can be made to force it.
```
Comment (mobileoverlord) [inline README.md]: > are set or a custom `fwup.conf` can be made to force it

Feels like there should be a comma here?
are set or, you could create a custom `fwup.conf` to force setting `nerves_autovalidate`.

File: README.md [hunk 4]
```diff
@@ -172,35 +173,50 @@ run:
...
+
+### Best effort automatic revert
+
+Unfortunately, the bootloader for some platforms like the Raspberry Pis makes it
+impossible to implement the above mechanism. The following strategy cannot
```
Comment (mobileoverlord) [inline README.md]: is it impossible or just very difficult?

File: README.md (hunk 1 above)
Comment (fhunleth) [reply README.md]: lol. Thanks for catching that.

File: README.md (hunk 4 above)
Comment (fhunleth) [reply README.md]: ok, fine, if you change the nerves_system_rpi to boot uboot, then you can do that. However, if you do that, then all of the documentation on the web about using config.txt to run dtc overlays won't work and then you need to fix that.

File: README.md [hunk 5]
```diff
@@ -172,35 +173,53 @@ run:
...
-Here's a simple alternative:
-
-1. After upgrading firmware, save that the next boot is the first one.
-1. On the reboot, if this is the first boot, record that the boot happened and
+Running this command manually is useful in development. Production use requires
+more work to protect against faulty upgrades.
+
+### Assisted firmware validation and automatic revert
+
+The Nerves firmware updates protect against update corruption and powerloss
```
Comment (mobileoverlord) [inline README.md]: > The Nerves firmware updates protect against update corruption

Should this be 

`Nerves firmware updates protect against update corruption`

**Pull request merged**